src/client.rs

```diff
@@ -559,14 +559,18 @@ impl Client {
         data: &[u8],
         digest: &str,
     ) -> Result<String> {
-        match self.push_blob_chunked(image_ref, data, digest).await {
-            Ok(url) => Ok(url),
-            Err(OciDistributionError::SpecViolationError(violation)) => {
-                warn!(?violation, "Registry is not respecting the OCI Distribution Specification when doing chunked push operations");
-                warn!("Attempting monolithic push");
-                self.push_blob_monolithically(image_ref, data, digest).await
+        if self.config.use_monolithic_push {
+            self.push_blob_monolithically(image_ref, data, digest).await
+        } else {
```

Comment:
Just a matter of personal taste, I would rewrite it get rid of the extra indentation caused by the `else` statement:

```rust
if self.config.use_monolithic_push {
  return self.push_blob_monolithically(image_ref, data, digest).await;
} 
```

Reply:
Great suggestion, I've adjusted the code accordingly 👍